Source/MaestroAdvection.cpp

```diff
@@ -727,14 +729,495 @@ Maestro::VelPred (const Vector<MultiFab>& utilde,
+                           force_mf.nComp(),
+                           BL_TO_FORTRAN_ANYD(u_mf[mfi]),
+                           BL_TO_FORTRAN_ANYD(v_mf[mfi]),
+                           BL_TO_FORTRAN_ANYD(w_mf[mfi]),
+                           BL_TO_FORTRAN_ANYD(Ipfx[mfi]),
+                           BL_TO_FORTRAN_ANYD(Imfx[mfi]),
+                           AMREX_INT_ANYD(domainBox.loVect()),
+                           AMREX_INT_ANYD(domainBox.hiVect()),
+                           bc_f, AMREX_REAL_ANYD(dx), dt, false,
+                           1,1,AMREX_SPACEDIM);
+                }
+            }
+
+            // y-direction
+            if (ppm_type == 0) {
+               // we're going to reuse Im here as slopey as it has the
+               // correct number of ghost zones
+#pragma gpu box(obx)
+               slopey_2d(AMREX_INT_ANYD(obx.loVect()),
+                      AMREX_INT_ANYD(obx.hiVect()),
+                      BL_TO_FORTRAN_ANYD(utilde_mf[mfi]),
+                      utilde_mf.nComp(),
+                      BL_TO_FORTRAN_ANYD(Imv[mfi]),Imv.nComp(),
+                      AMREX_INT_ANYD(domainBox.loVect()),
+                      AMREX_INT_ANYD(domainBox.hiVect()),
+                      3,bc_f,AMREX_SPACEDIM,1);
+
+            } else {
+
+#pragma gpu box(obx)
+               ppm_3d(AMREX_INT_ANYD(obx.loVect()),
+                      AMREX_INT_ANYD(obx.hiVect()),
+                      BL_TO_FORTRAN_ANYD(utilde_mf[mfi]),
+                      utilde_mf.nComp(),
+                      BL_TO_FORTRAN_ANYD(u_mf[mfi]),
+                      BL_TO_FORTRAN_ANYD(v_mf[mfi]),
+                      BL_TO_FORTRAN_ANYD(w_mf[mfi]),
+                      BL_TO_FORTRAN_ANYD(Ipv[mfi]),
+                      BL_TO_FORTRAN_ANYD(Imv[mfi]),
+                      AMREX_INT_ANYD(domainBox.loVect()),
+                      AMREX_INT_ANYD(domainBox.hiVect()),
+                      bc_f, AMREX_REAL_ANYD(dx), dt, false,
+                      2,2,AMREX_SPACEDIM);
+
+              if (ppm_trace_forces == 1) {
+#pragma gpu box(obx)
+                ppm_3d(AMREX_INT_ANYD(obx.loVect()),
+                    AMREX_INT_ANYD(obx.hiVect()),
+                    BL_TO_FORTRAN_ANYD(force_mf[mfi]),
+                    force_mf.nComp(),
+                    BL_TO_FORTRAN_ANYD(u_mf[mfi]),
+                    BL_TO_FORTRAN_ANYD(v_mf[mfi]),
+                    BL_TO_FORTRAN_ANYD(w_mf[mfi]),
+                    BL_TO_FORTRAN_ANYD(Ipfy[mfi]),
+                    BL_TO_FORTRAN_ANYD(Imfy[mfi]),
+                    AMREX_INT_ANYD(domainBox.loVect()),
+                    AMREX_INT_ANYD(domainBox.hiVect()),
+                    bc_f, AMREX_REAL_ANYD(dx), dt, false,
+                    2,2,AMREX_SPACEDIM);
+              }
+            }
+
+            // z-direction
+            if (ppm_type == 0) {
+               // we're going to reuse Im here as slopey as it has the
+               // correct number of ghost zones
```

Comment:
comment says Im, but the code is using Imw?

Reply:
That's probably because I copied and pasted from the 2d case. For the x-component, it's Ipu, for the y-component it's Imv, and for the z-component it's Imw